src/WebJobs.Script.WebHost/ContainerManagement/LinuxFunctionExecutionActivityPublisher.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.WebHost.Management;
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.ContainerManagement
+{
+    public class LinuxFunctionExecutionActivityPublisher : IHostedService, IDisposable, ILinuxFunctionExecutionActivityPublisher
+    {
+        private const int FlushIntervalMs = 5 * 1000; // 5 seconds
+
+        private readonly IMeshInitServiceClient _meshInitServiceClient;
+        private readonly ILogger<LinuxFunctionExecutionActivityPublisher> _logger;
+        private readonly int _flushIntervalMs;
+        private readonly ConcurrentQueue<ContainerFunctionExecutionActivity> _queue;
+        private readonly IOptionsMonitor<StandbyOptions> _standbyOptions;
+        private IDisposable _standbyOptionsOnChangeSubscription;
+        private DateTime _lastHeartBeatTime = DateTime.MinValue;
+        private Timer _timer;
+        private int _flushInProgress;
+
+        public LinuxFunctionExecutionActivityPublisher(IOptionsMonitor<StandbyOptions> standbyOptions,
+            IMeshInitServiceClient meshInitServiceClient, IEnvironment environment,
+            ILogger<LinuxFunctionExecutionActivityPublisher> logger, int flushIntervalMs = FlushIntervalMs)
+        {
+            if (!environment.IsLinuxConsumption())
+            {
+                throw new NotSupportedException(
+                    $"{nameof(LinuxFunctionExecutionActivityPublisher)} is available in Linux consumption environment only");
+            }
+
+            _standbyOptions = standbyOptions ?? throw new ArgumentNullException(nameof(standbyOptions));
+            _meshInitServiceClient = meshInitServiceClient;
+            _logger = logger;
+            _flushIntervalMs = flushIntervalMs;
+            _timer = new Timer(OnTimer, null, Timeout.Infinite, Timeout.Infinite);
+            _queue = new ConcurrentQueue<ContainerFunctionExecutionActivity>();
+            _flushInProgress = 0;
+        }
+
+        private void Start()
+        {
+            _logger.LogInformation($"Starting {nameof(LinuxFunctionExecutionActivityPublisher)}");
+
+            // start the timer by setting the due time
+            SetTimerInterval(_flushIntervalMs);
+        }
+
+        private void OnStandbyOptionsChange()
+        {
+            _logger.LogInformation($"Triggering {nameof(OnStandbyOptionsChange)}");
```

Comment:
nit: I wonder if instead of this log here, you shouldn't just have a log in the Start method when starting